app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,28 @@
+# require 'pry'
+class SessionsController < ApplicationController
+
+  def new
+    @user
+  end
+
+  def create
+    auth_hash = request.env['omniauth.auth']
+    if auth_hash[:uid]
+      @user = User.find_or_create_from_omniauth(auth_hash)
+      if @user
+        session[:user_id] = @user.id
+        redirect_to root_path
+      else
+        redirect_to root_path, notice: "Failed to save the user"
+      end
+    else
+      redirect_to root_path, notice: "Failed to authenticate"
+    end
+  end
+
+  def destroy
+    session.delete :user_id
+    redirect_to root_path
+  end
```

Comment:
What's an alternate way that you would tend to do it? If you don't mind? :-)

On Tue, May 24, 2016 at 2:57 PM, Riley Spicer <notifications@github.com>
wrote:

> In app/controllers/sessions_controller.rb
> <https://github.com/Ada-C5/tunes-takeout/pull/5#discussion_r64481177>:
>
> > +      @user = User.find_or_create_from_omniauth(auth_hash)
> > +      if @user
> > +        session[:user_id] = @user.id
> > +        redirect_to root_path
> > +      else
> > +        redirect_to root_path, notice: "Failed to save the user"
> > +      end
> > +    else
> > +      redirect_to root_path, notice: "Failed to authenticate"
> > +    end
> > +  end
> > +
> > +  def destroy
> > +    session.delete :user_id
> > +    redirect_to root_path
> > +  end
>
> Cool, I hadn't seen this way to delete things from the session hash
> before. (So many ways to do each thing in Ruby lol)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Ada-C5/tunes-takeout/pull/5/files/44c4d956d96ca833d6958fa91f68e369fa694b1c#r64481177>
>